bpf/lib/nat.h

```diff
@@ -1220,9 +1231,7 @@ snat_v6_nat_handle_mapping(struct __ctx_buff *ctx,
 		if (IS_ERR(ret))
 			return ret;
 
-		set_v6_rtuple(tuple, *state, &rtuple);
-		*state = snat_v6_lookup(&rtuple);
-		if (*state)
+		if (rstate)
```

Comment:
(commenting here as the diff is cleaner)

to clarify - when the RevSNAT entry is missing, we
1. delete the SNAT entry
2. let `snat_v6_new_mapping()` create a new SNAT / RevSNAT pair of entries.

In a busy node, won't `snat_v6_new_mapping()` then end up selecting a different source port for this connection? Won't this result in disrupted connectivity?

I would have expected that we preserve the SNAT entry, and merely re-create the RevSNAT entry from the available information. Why is that not possible?

Reply:
Good point. Updated!